---
title: "report"
author: "Vladimir Kiselev"
date: "10 February 2015"
output:
  pdf_document:
    number_sections: yes
    toc: yes
---

```{r, echo=FALSE, include=FALSE, cache=TRUE}
source("functions.R")
```

# Polysome data

## Pre-processing (process_data() function)

By removing HTML header in the original **original-data/p28_expression_genes.txt** file I obtained **original-data/p28_expression_genes_edited.txt** file, which I used as the main data file. I imported data from this file into R, then removed **Poly_L_2_17** sample from the data and then created data tables for statistical tests (**files/data-matrix.rds**), clustering and preliminary analysis (**files/data-matrix-norm.rds**), plotting (**files/plot-table.rds**) and annotating (**files/ann-table.rds**; annotation data was downloaded from Ensembl Biomart).

## Correlations between samples
All 156 Polysome sample were correlated against each other. The correlation between samples is quite low (even for technical replicates).

```{r, echo=FALSE, cache=TRUE, fig.width=10, fig.height=10}
library(gplots)
cor_heatmap <- function(count.matrix, name) {
        # plot a correlation matrix from a count matrix
        # calculate pearson's correlation coefficients
        cor.matrix <- cor(count.matrix, method = "pearson")
        # plot correlation matrix in a file with 'name'
        #   plot(1)
        heatmap.2(cor.matrix, Rowv = FALSE, Colv = FALSE, dendrogram = "none", col=bluered(9), breaks = 10, trace = "none", keysize = 0.5)
}
d <- readRDS("files/data-matrix-norm.rds")

cols <- colnames(d[,2:157])
j <- 4
for(i in 1:13){
  cols[grep(paste0("LUL1._e", i, "_"), cols)] <- paste0("L_", j)
  j <- j + 1
}
j <- 4
for(i in 14:26){
  cols[grep(paste0("LUL1._e", i, "_"), cols)] <- paste0("LE_", j)
  j <- j + 1
}
j <- 4
for(i in 27:39){
  cols[grep(paste0("LUL1._e", i, "_"), cols)] <- paste0("LEKU_", j)
  j <- j + 1
}
colnames(d)[2:157] <- cols
tmp <- as.matrix(d[,2:157])
cor_heatmap(tmp[,order(colnames(tmp))], "polysome")
```

## Principal Component Analysis (PCA) of Polysome data
PCA analysis of of the Polysome shows that ~37% of data variance is explained by the first 3 principal components components.

```{r, echo=FALSE, cache=TRUE, fig.width=4, fig.height=5}
library(ggplot2)
res <- prcomp(d[,2:157])
screeplot(res)
```
```{r, echo=FALSE, cache=TRUE, fig.width=4, fig.height=5}
data <- as.data.frame(res$rotation[,1:3])
data$cond <- rownames(data)
data$cond <- unlist(lapply(strsplit(data$cond, "\\."), function(x){return(x[1])}))
data$pf <- as.numeric(unlist(lapply(strsplit(data$cond, "_"), function(x){return(x[2])})))
data$cond <- unlist(lapply(strsplit(data$cond, "_"), function(x){return(x[1])}))
ggplot(data, aes(PC1,PC2)) +
geom_point(aes(shape = cond, color = as.factor(pf))) +
scale_size(range = c(3, 6)) +
theme_bw()
```
```{r, echo=FALSE, cache=TRUE, fig.width=4, fig.height=5}
ggplot(data, aes(PC2,PC3)) +
geom_point(aes(shape = cond, color = as.factor(pf))) +
scale_size(range = c(3, 6)) +
theme_bw()
```
```{r, echo=FALSE, cache=TRUE, fig.width=4, fig.height=5}
ggplot(data, aes(PC1,PC3)) +
geom_point(aes(shape = cond, color = as.factor(pf))) +
scale_size(range = c(3, 6)) +
theme_bw()
```

## Statistical analysis

To identify changes in polysome fraction profile DESeq2 package was used (it can be used with any sequencing count data). First, a GLM (read.counts ~ condition + polysome.fraction + condition:polysome.fraction) was estimated for polysome data and, second, the likelihood ratio test was performed between the original and a reduced (read.counts ~ polysome.fraction) GLMs.

Number of genes in each genes set:

```{r, echo=FALSE, cache=TRUE}
initialize_gene_sets()
```

LE vs L: 4755

LEKU vs L: 4770

LE vs LEKU: 1402


For comparison, previous results from ANOVA analysis (will not use them anymore):


LE vs L: 4080

LEKU vs L: 2941

LE vs LEKU: 1389


Top 10 most significant genes in LE vs L comparison:

```{r, echo=FALSE, cache=TRUE}
head(res.l.le.deseq[order(res.l.le.deseq$padj),], 10)
```

```{r, echo=FALSE, cache=TRUE, fig.width=8, fig.height=5}
plot_genes <- function(gene.names){
        plot.data <- get_plot_data(gene.names)
        limits <- aes(ymax = mean_value + sd_value, ymin = mean_value - sd_value)
        ggplot(plot.data, aes(as.numeric(pf), mean_value, group = cond, color = cond)) +
                geom_line() + facet_wrap(~ gene_name, scale = "free_y") +
                geom_errorbar(limits, width = 0.25) +
                labs(x = "Polysome fraction", y = "Read counts") +
                theme_bw()
}
plot_genes(head(res.l.le.deseq[order(res.l.le.deseq$padj),], 10)$ensembl_gene_id)
```


Top 10 most significant genes in LEKU vs L comparison:

```{r, echo=FALSE, cache=TRUE}
head(res.l.leku.deseq[order(res.l.leku.deseq$padj),], 10)
```

```{r, echo=FALSE, cache=TRUE, fig.width=8, fig.height=5}
plot_genes(head(res.l.leku.deseq[order(res.l.leku.deseq$padj),], 10)$ensembl_gene_id)
```


Top 10 most significant genes in LEKU vs LE comparison:

```{r, echo=FALSE, cache=TRUE}
head(res.le.leku.deseq[order(res.le.leku.deseq$padj),], 10)
```

```{r, echo=FALSE, cache=TRUE, fig.width=8, fig.height=5}
plot_genes(head(res.le.leku.deseq[order(res.le.leku.deseq$padj),], 10)$ensembl_gene_id)
```

# RNA-Seq data

## Correlation between samples

Correlations of read numbers in 12 RNA-Seq samples are very high (even between different conditions):

```{r, echo=FALSE, message=FALSE, cache=TRUE, fig.width=10, fig.height=10}
library(DESeq2)
count_matrix_from_files <- function(files, normalized) {
        # this function create a count matrix from raw htseq read counts files
        # it uses DESeq2 library to construct the matrix
        # create a design matrix for DESeq2
        sample.name <- sapply(strsplit(files, "_trimmed"), "[[", 1)
        condition <- sapply(strsplit(sample.name, "_"), "[[", 1)
        samples <- data.frame(sample.name = sample.name,
        file.name = paste0("htseq-count/", files),
        condition = condition)
        samples$condition <- factor(samples$condition,
        levels = unique(samples$condition))
        # import data from files using 'samples' design matrix
        cds <- DESeqDataSetFromHTSeqCount(sampleTable = samples,
        directory = ".", design = ~ condition)
        # if normalized is TRUE then the count matrix is normalized by library size
        # using estimateSizeFactors function of DESeq2
        if (normalized) {
                cds <- estimateSizeFactors(cds)
                return(counts(cds, normalized = TRUE))
        } # otherwise return the raw matrix
        else {return(counts(cds))}
}

files <- list.files("htseq-count/", full.names = FALSE)
count.matrix <- count_matrix_from_files(files, T)
colnames(count.matrix) <- unlist(lapply(strsplit(colnames(count.matrix), "_"), function(x){return(x[2])}))
count.matrix <- count.matrix[,order(colnames(count.matrix))]
cor_heatmap(count.matrix, "rna-seq")
```

## Statistical analysis
For differential expression analysis I used already processed and aligned RNA-Seq data (BAM files provided by Manuel). BAM files were converted into SAM files using Samtools (**samtools.sh** script). Read counts in gene features were calculated using HT-Seq tool and **Mus\_musculus.GRCm38.78.gtf** feature file downloaded from Ensembl (**htseq.sh** script). Differential expression analysis was performed using DESeq2 package based on the obtained read count files. Condition comparisonsLE vs L, LE vs LEKU and LEKU vs L were tested for differential expression. 0.01 threshold of adjusted p-values was used as level of significance.

Number of genes in each genes set:


LE vs L: 8109

LEKU vs L: 7283

LE vs LEKU: 4534


A summarized Venn diagram showing an overlap of these three gene sets:


```{r, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE, fig.width=9, fig.height=6, fig.align='center'}
library(Vennerable)
venn <- function(list, is.weights) {
  v <- Venn(list)
  plot(v, doWeights = is.weights)
}
venn(list("LE-vs-L" = rownames(le), "LEKU-vs-L" = rownames(leku),
        "LE-vs-LEKU" = rownames(le.leku)), T)
```
